web/src/reactions.js

```diff
@@ -78,6 +80,10 @@ function update_ui_and_send_reaction_ajax(message_id, reaction_info) {
     const operation = has_reacted ? "remove" : "add";
     const reaction = create_reaction(message_id, reaction_info);
 
+    if (view.loading_reactions_ids[local_id]) {
+        return;
+    }
```

Comment:
So I think this change is a good direction, but `local_id` is the wrong key. `local_id` defines a single reaction symbol, but not the message, so if you click `+1` on two different messages i quick succession, this change will ignore the second.

I think you need to use some encoding of a `(message_id, local_id)` tuple as they key here.

Reply:
Thanks for the example. You absolutely right, it would have caused issues if we had only encoded the `local_id`. I will update the logic accordingly.